workflow/controller/operator.go

```diff
@@ -554,6 +554,13 @@ func (woc *wfOperationCtx) persistUpdates(ctx context.Context) {
 		woc.controller.hydrator.HydrateWithNodes(woc.wf, nodes)
 	}
 
+	// The workflow returned from wfClient.Update doesn't have a TypeMeta associated
+	// with it, so copy from the original workflow.
+	woc.wf.TypeMeta = woc.orig.TypeMeta
```

Comment:
Might be best to do this when we make to copy.

Reply:
Yeah, I don't like this particular solution either, but fixing `wfClient.Update` to return a workflow with a populated `TypeMeta` seems out of scope for this PR.